framework/components/ADataTable/ADataTable.js

```diff
@@ -1,18 +1,32 @@
 import PropTypes from "prop-types";
-import React, {forwardRef} from "react";
+import React, {forwardRef, useState} from "react";
 
 import AIcon from "../AIcon";
 import ASimpleTable from "../ASimpleTable";
 import "./ADataTable.scss";
 
-const ADataTable = forwardRef(
-  ({className: propsClassName, headers, items, onSort, sort, ...rest}, ref) => {
-    let className = "a-data-table";
+const TableHeader = (props) => <th role='columnheader' className='a-data-table__header' {...props} />;
+const TableRow = (props) => <tr role='row' className='a-data-table__row' {...props} />;
+const TableCell = (props) => <td role='cell' className='a-data-table__cell' {...props} />;
 
+const ADataTable = forwardRef(
+  ({className: propsClassName, expandable, headers, items, onSort, sort, ...rest}, ref) => {
+    const [expandedRows, setExpandedRows] = useState({});
+    const ExpandableComponent = expandable?.component;
+    let className = 'a-data-table';
+    if (ExpandableComponent) {
+      className += '--expandable'
```

Comment:
It's still `a-data-table`, and it's also `a-data-table--expandable`. The modifier doesn't replace the root class name, but adds on to it. So className would need to be `a-data-table a-data-table--expandable` at this point